bin/ch-completion.bash

```diff
@@ -0,0 +1,403 @@
+            return 0
+            ;;
+        -t )
+            # We can't autocomplete a tag, so we're not even gonna allow
+            # autocomplete after this option.
+            COMPREPLY=()
+            return 0
+            ;;
+        *)
+            # Autocomplete to context directory, common opt, or build-specific opt
+            # --force can take “fakeroot” or “seccomp” as an argument, or no
+            # argument at all. To account for this, we add those two arguments
+            # to the list of compgen suggestions, which allows compgen to
+            # autocomplete to “fakeroot,” “seccomp,” or anything that could
+            # logically follow “--force” with no argument.
+            if [[ "$prev" == "--force" ]]; then
+                extras+="$extras fakeroot seccomp"
+            fi
+            COMPREPLY=( $(compgen -W "$_image_build_opts $extras"  -- "$cur") )
+            # By default, “complete” adds a space after each completed word.
+            # This is incredibly inconvenient when completing directories and
+            # filepaths, so we enable the “nospace” option. We want to make sure
+            # that this option is only enabled if there are valid path
+            # completions for “cur,” otherwise spaces would never be added after
+            # a completed word, which is also inconveninet.
+            if [[ -n "$(compgen -d -S / -- "$cur")" ]]; then
+                compopt -o nospace
+                COMPREPLY+=( $(compgen -d -S / -- "$cur") )
+            fi
+            ;;
+        esac
+        ;;
+    build-cache )
+        COMPREPLY=( $(compgen -W "--reset --gc --tree --dot" -- "$cur") )
+        ;;
+    delete | list )
+        if [[ "$sub_cmd" == "list" ]]; then
+            extras+="$extras -l --long"
+        fi
+        # The following check seems to protects against trying to ls a
+        # non-existent directory, and fixes a bug where the completion function
+        # initialzes an empty storage directory.
+        if [[ -f "$strg_dir" && -n "$(ls "$strg_dir/img")" ]]; then
+            COMPREPLY=( $(compgen -W "$(ls "$strg_dir/img" | sed 's/+/:/g' | sed 's/%/\//g') $extras" -- "$cur") )
+            __ltrim_colon_completions "$cur"
+        fi
+        ;;
+    gestalt )
+        COMPREPLY=( $(compgen -W "bucache bucache-dot python-path
+                                  storage-path" -- "$cur") )
+        ;;
+    import )
+        # Complete dirs and files matching the globs “*.tar.*” and “*.tgz”
+        # (a.k.a. tarballs).
+        COMPREPLY+=( $(_compgen_filepaths -X "!*.tar.* !*tgz" "$cur") )
+        if [[ ${#COMPREPLY} -gt 0 ]]; then
+            compopt -o nospace
+        fi
+        ;;
+    push )
+        if [[ "$prev" == "--image" ]]; then
+            compopt -o nospace
+            COMPREPLY=( $(compgen -d -S / -- "$cur") )
+            return 0
+        # The following check seems to fix a bug where the completion function
+        # initialzes an empty storage directory.
```

Comment:
This is the second time this comment appears. I think I would instead put a general comment at the top saying you don't run `ch-image list` or any other commands that have side effects like initializing the storage directory.